blockdag/mining.go

```diff
@@ -0,0 +1,133 @@
+package blockdag
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/kaspanet/kaspad/txscript"
+	"github.com/kaspanet/kaspad/util"
+	"github.com/kaspanet/kaspad/util/subnetworkid"
+	"github.com/kaspanet/kaspad/wire"
+	"sort"
+	"time"
+)
+
+// BlockForMining returns a block with the given transactions
+// that points to the current DAG tips, that is valid from
+// all aspects except proof of work.
+func (dag *BlockDAG) BlockForMining(transactions []*util.Tx, useMinimalTime bool) (*wire.MsgBlock, error) {
```

Comment:
Is useMinimalTime only there for tests?

If so, I'd rather it not be parameter of dag.BlockForMining. Instead, get the block with timestamp according to dag.timeSource, and than modify it for what the tests needs.